Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/DefaultRiskLevels.kt

```diff
@@ -216,16 +159,313 @@ class DefaultRiskLevels @Inject constructor() : RiskLevels {
+                .reportTypeOffsetConfirmedTest
+            else -> throw UnknownReportTypeException()
+        }
+
+        val infectiousnessOffset = when (exposureWindow.infectiousness) {
+            Infectiousness.HIGH -> transmissionRiskLevelEncoding
+                .infectiousnessOffsetHigh
+            else -> transmissionRiskLevelEncoding
+                .infectiousnessOffsetStandard
+        }
+
+        return reportTypeOffset + infectiousnessOffset
+    }
+
+    private fun dropDueToTransmissionRiskLevel(
+        transmissionRiskLevel: Int,
+        transmissionRiskLevelFilters: List<RiskCalculationParametersOuterClass.TrlFilter>
+    ) =
+        transmissionRiskLevelFilters.any {
+            it.dropIfTrlInRange.inRange(transmissionRiskLevel)
+        }
+
+    private fun determineWeightedSeconds(
+        exposureWindow: ExposureWindow,
+        minutesAtAttenuationWeight: List<RiskCalculationParametersOuterClass.MinutesAtAttenuationWeight>
+    ): Double =
+        exposureWindow.scanInstances.fold(.0) { seconds, scanInstance ->
+            val weight =
+                minutesAtAttenuationWeight
+                    .filter { it.attenuationRange.inRange(scanInstance.typicalAttenuationDb) }
+                    .map { it.weight }
+                    .firstOrNull() ?: .0
+            seconds + scanInstance.secondsSinceLastScan * weight
+        }
+
+    private fun determineRiskLevel(
+        normalizedTime: Double,
+        timeToRiskLevelMapping: List<RiskCalculationParametersOuterClass.NormalizedTimeToRiskLevelMapping>
+    ) =
+        timeToRiskLevelMapping
+            .filter { it.normalizedTimeRange.inRange(normalizedTime) }
+            .map { it.riskLevel }
+            .firstOrNull()
+
+    override fun calculateRisk(
+        exposureWindow: ExposureWindow
+    ): RiskResult? {
+        if (dropDueToMinutesAtAttenuation(exposureWindow, appConfig.minutesAtAttenuationFilters)) {
+            Timber.d(
+                "%s dropped due to minutes at attenuation filter",
+                exposureWindow
+            )
+            return null
+        }
+
+        val transmissionRiskLevel = determineTransmissionRiskLevel(
+            exposureWindow,
+            appConfig.transmissionRiskLevelEncoding
+        )
+
+        if (dropDueToTransmissionRiskLevel(transmissionRiskLevel, appConfig.transmissionRiskLevelFilters)) {
+            Timber.d(
+                "%s dropped due to transmission risk level filter, level is %s",
+                exposureWindow,
+                transmissionRiskLevel
+            )
+            return null
+        }
+
+        val transmissionRiskValue =
+            transmissionRiskLevel * appConfig.transmissionRiskLevelMultiplier
+
+        Timber.d(
+            "%s's transmissionRiskValue is: %s",
+            exposureWindow,
+            transmissionRiskValue
+        )
+
+        val weightedMinutes = determineWeightedSeconds(
```

Comment:
Please add an explicit type here. If we expect a fractional number, also force it by specifying `60f` or `60.0`

Reply:
done in #1656